### DISCUSSION TITLE
NOBODY IS ACCEPTING MY FILES

### DISCUSSION THREAD

**Discussion by AllanOricil** [filecoin-project/lotus#7838] [General Q&A]
### Checklist

- [X] This is **not** a security-related bug/issue. If it is, please follow please follow the [security policy](https://github.com/filecoin-project/lotus/security/policy).
- [X] This is **not** a question or a support request. If you have any lotus related questions, please ask in the [lotus forum](https://github.com/filecoin-project/lotus/discussions).
- [X] This is **not** a new feature request. If it is, please file a [feature request](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Ffeature&template=feature_request.yml) instead.
- [X] This is **not** an enhancement request. If it is, please file a [improvement suggestion](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Fenhancement&template=enhancement.yml) instead.
- [X] I **have** searched on the [issue tracker](https://github.com/filecoin-project/lotus/issues) and the [lotus forum](https://github.com/filecoin-project/lotus/discussions), and there is no existing related issue or discussion.
- [X] I am running the [`Latest release`](https://github.com/filecoin-project/lotus/releases), or the most recent RC(release canadiate) for the upcoming release or the dev branch(master), or have an issue updating to any of these.
- [X] I did not make any code changes to lotus.

### Lotus component

- [ ] lotus daemon - chain sync
- [ ] lotus miner - mining and block production
- [ ] lotus miner/worker - sealing
- [ ] lotus miner - proving(WindowPoSt)
- [ ] lotus miner/market - storage deal
- [ ] lotus miner/market - retrieval deal
- [ ] lotus miner/market - data transfer
- [ ] lotus client
- [ ] lotus JSON-RPC API
- [ ] lotus message management (mpool)
- [X] Other

### Lotus Version

```text
NA
```

### Describe the Bug

Nome of the Plus Providers are accepting my files. I tried around 15 already with no sucess. Sometimes they say 0 attoFils and then when I dispatch a deal they come back saying something like: 0.00000...2 Fils something...
And the deal fail.
Or Sometimes they say they are not accepting online deals. WTF! If they are on the network they MUST accept online deals. What are these guys doing? Adding files themselves to mine Filecoin? I don't understand how they got 100 trust if they are not accepting online deals. 
Or Sometimes the deal is pending something. Why do I have to wait the provider to MANUALLY accept something if we already agreed the terms? I've been waiting for almost a day and the file is still not available to be retrieved.
If you want to make it work, you have to make things to go faster in the provider side. Otherwise what is the point? Long wait time is a huge trade-off that will make everyone go away from this.

### Logging Information

```text
NA
```

### Repo Steps

NA

**Comment by Reiers** [comment]
**Hi @AllanOricil** 

Thanks for the report. This is not a Lotus issue.

Moving this issue to [Discussions](https://github.com/filecoin-project/lotus/discussions) for help and troubleshooting.

Closing ticket.

**Comment by rjan90** [comment]
Hi @AllanOricil!

There are a number of reasons why Storage Providers might not accept your files. First of all, have you got a DataCap so you can store files with the verified price? [https://docs.filecoin.io/store/filecoin-plus/#using-datacap](https://docs.filecoin.io/store/filecoin-plus/#using-datacap).

This text might indicated that you where trying to make a storage-deal with unverified data:
>Sometimes they say 0 attoFils and then when I dispatch a deal they come back saying something like: 0.00000...2 Fils something... And the deal fail. 

Secondly, storage-providers might block clients they do not know who is, or that they do not have on their whitelist. You can reach out to a lot of the providers in the #fil-deal-market channel on Slack to troubleshoot why your deal fails.

>I've been waiting for almost a day and the file is still not available to be retrieved.
You should be able to retrieve the file right away after making a deal. So this might need some troubleshooting. 

Lastly I would like to encourage you to have a more respectful framing of your questions. Writing in all caps, and statements that can be interpreted by some as rude, might make people not wanting to answer your questions.

**Reply by AllanOricil** [reply]
The words in CAPS are for enphasys, not to be rude. Or when I write a word in caps is because it is an acronym. like WTF. If I want to be rude writting in caps I will say bad words/phrases.

**Comment by AllanOricil** [comment]
This really upsets me. I literally have been trying to store a single file using this [tutorial](https://lotus.filecoin.io/docs/tutorials/store-and-retrieve/#run-a-lotus-lite-node) and it does not work because none of the providers in the chain wants to accept files. So, what are these people doing? You MUST enforce them to accept files IF I PAY THEM WHAT THEY WANT, because I lost money with some trials and none worked.

![image](https://user-images.githubusercontent.com/55927613/147083618-39d0dda6-5946-4e21-ba52-4c285cdd6a5c.png)

**Reply by AllanOricil** [reply]
Why these possible errors even exist?  
1 - deal rejected: miner is not considering online storage deals
2 - Price below acceptance for such deal : 0.0000002 FIL | For any questions you can reach me on Slack @stuberman

On 1 I paid what they wanted, I lost money, and then they can reject? WTF! This makes no sense. We had a DEAL, and they can reject? and I will lose money? Makes no sense

On 2 I also paid what the deal was asking, and then he returned saying he won't accept it? Why? The DEAL went through the validations. It was saying 0 attoFil. How the hell he wants 0.0000002 FIL? 

I just don't think it is fair I keep losing money with these trials while these Providers can reject the file. Nobody will ever use it if you don't enforce some rules in the chain.

**Reply by rjan90** [reply]
1. There are multiple reasons why a storage-provider might not take on your storage-deal at this exact time. Upgrades, full sealing pipeline, etc, are just some of the reasons. And no, you won't loose any money if the deal is rejected.

2. The error indicates that the price you where trying to make a deal with the storage-provider was below his acceptance price. His slack-name is added if you want to discuss further why it was rejected. And no, you don't loose any money, when the deal errors out like this as well.

**Reply by AllanOricil** [reply]
> 1. There are multiple reasons why a storage-provider might not take on your storage-deal at this exact time. Upgrades, full sealing pipeline, etc, are just some of the reasons. And no, you won't loose any money if the deal is rejected.
> 2. The error indicates that the price you where trying to make a deal with the storage-provider was below his acceptance price. His slack-name is added is you want to discuss further why it was rejected. And no, you don't loose any money, when the deal errors out like this as well.

My wallet balance was reduced after I accepted the Deal. Is this money going back to my wallet later?

**Reply by rjan90** [reply]
> My wallet balance was reduced after I accepted the Deal. Is this money going back to my wallet later?

Yes, they will.

**Reply by AllanOricil** [reply]
Also, could you share a Provider that would allow me to continue through this tutorial: 

https://lotus.filecoin.io/docs/tutorials/store-and-retrieve/#run-a-lotus-lite-node

Im stuck in the Retrieval part because I can't upload a file.

Also, sorry if I sound rude. It is just that I wanted to learn this and for the past 3 days I've had lots of problems. And I really get frustrated when I go through tutorials and they don't work, or when I get stuck in some part because of something I did not do. But I really did not want to sound rude.

**Reply by rjan90** [reply]
I would recommend you join on Slack, it might be a bit easier to troubleshoot there (you can reach out to me there and I can help you further, DM @phi-rjan).

And I would also say that you bring up a lot of valid points with regards to the UX making deals using that tutorial currently. And I have raised awareness about that.

There are also easier tools then using Lotus-CLI if all you want is just storing a file. For example: [https://estuary.tech](https://estuary.tech).

**Reply by AllanOricil** [reply]
I will join slack. Where can I find the link?

**Reply by rjan90** [reply]
At the bottom of [http://filecoin.io/](http://filecoin.io/) there is a link to Slack 🙂

**Comment by AllanOricil** [comment]
Also, can you add a filter on the Filecoin Plus providers list to remove Providers that are not considering online storage deals. "deal rejected: miner is not considering online storage deals". 

And throw an error in the client when we try to start a deal with such providers.

**Reply by jennijuju** [reply]
@dkkapur

**Comment by dd45e640b42e6da7da96faee3996ef7c** [comment]
> Why do I have to wait the provider to MANUALLY accept something if we already agreed the terms?

on a note: your deals counts as accepted as soon as the proposal is published on chain -> has both clients and storage providers signatures on it. maybe your offers are not economically feasible to storage providers?

**Comment by dd45e640b42e6da7da96faee3996ef7c** [comment]
your deals are too small / do not pay enough to be considered economically reasonable i think

check the gas burn for publishdeals messages on chain or an explorer like filfox.info to see what the baseline costs are for publishing deals on chain for a storage provider - everything below that will cost the storage provider money --> they do not take your deals

**Reply by AllanOricil** [reply]
But I stil think it is weird to see it working this way. Like, when I ran `lotus client deal` the miner deal appears in the screen and I have to accept its terms. That deal information was configured by the Miner. I accept it and they can still not accept to store my file? Like, they proposed a Deal which they can deny? 
Imagine you negotiating a contract with a customer. The customer come to you and give you their proposal. You, then accept the deal, and both sign the contract. Now you are employed. If the customer wants to expire your contract before the agreed deadline, they have to pay you something. 
Now in the filecoin after accepting the Deal that the Miner proposed, they can still reject you. In this time you spent your time and money because you thought everything was Ok. This time won't come back, and you will have to spend more time comming after a new Miner, that could also reject the Deal they propose. This is unfair. 
I think Miners in the network can't say what or not they will accept after they configure the Deal.

**Reply by dd45e640b42e6da7da96faee3996ef7c** [reply]
>  I have to accept its terms.

you do not have to. it's your choice who to store with.

> Like, they proposed a Deal which they can deny?

it's a non binding one way information channel you are querying - the miner is not bound to it, nor are you.

> Imagine you negotiating a contract with a customer. The customer come to you and give you their proposal. You, then accept the deal, and both sign the contract. Now you are employed. If the customer wants to expire your contract before the agreed deadline, they have to pay you something.

no one signed anything until the storage client send a proposal to the storage provider. the deal is initiated by the storage client - not the storage provider. as soon as the deal is published on chain (at costs for the provider) the deal is agreed on. before that it's nothing but off chain talk

> Now in the filecoin after accepting the Deal that the Miner proposed

this is a misunderstanding. the miners to not offer deals. the client does.

**Reply by dd45e640b42e6da7da96faee3996ef7c** [reply]
> I think Miners in the network can't say what or not they will accept after they configure the Deal.

they do not do that. the client initiates a deal by sending a singed proposal, then sends the data and the miner compares that data he received to the data in the deal proposal, then the miner signs the proposal and send it to the chain. exactly at this point you have a valid filecoin deal.

**Reply by AllanOricil** [reply]
Can Miners also decide to one day stop their business and delete all my files? If so, Im forced to store my files in multiple Miners to decrease the chances of losing my backup. But I will have to sleep thinking that all can be lost. To remove this fear I must use centralized Cloud Providers. So, it is not worthy using FIlecoin at all. There has to be a way of making it profitable to Miners and at the same time give End Users relief that they can trust in this network and their files are safe, make it stupidly cheaper than AWS/Google Cloud/Azure/Alibaba Cloud, and finally as fast as those cloud provider are. Of course there will be trade-offs you will have to choose, or make it 100% visible so the End User can choose.

**Reply by dd45e640b42e6da7da96faee3996ef7c** [reply]
the "ask protocol" - the part where you ask a miner what he might store is under revision and a v2 version in the making that address some of your concerns. but it will not change the logic of an ask always being a non binding piece of information. the flow

- client signs a proposal
- miner sings the proposal
- deal gets published on chain (with both signatures)

**Reply by AllanOricil** [reply]
> > I think Miners in the network can't say what or not they will accept after they configure the Deal.
> 
> they do not do that. the client initiates a deal by sending a singed proposal, then sends the data and the miner compares that data he received to the data in the deal proposal, then the miner signs the proposal and send it to the chain. exactly at this point you have a valid filecoin deal.

Wait, Of course they can deny my Deal. I know that because some of my Deals were sending them exactly what they accept as File Size and FIL, and they denied to store it. As an End User, I don't want to keep having the Door shut in my face all the time. 

Maybe you should make Miners to configure the Deals with everything they are expecting. In a way that they are clear with what they want. This Deal information will then appear to the Client, for instance when I run `lotus client deal` and pass that Miners Id. The client will know upfront what is the Necessary in terms of Money and File size he must store there. The Client has the option to Negotiate with the Miner. But if the does not want to do so, he just accepts the Deal the Miner asked. The Miner receives the file and stores it at any cost. The Client waits for the ACK. The Client sleeps well.

**Reply by dd45e640b42e6da7da96faee3996ef7c** [reply]
>  Can Miners also decide to one day stop their business and delete all my files?

yes, if you paid for the storage service you'll get your remaining money back. miners pay hefty fees for not storing data they promised to stored - to the burn actor, not the client

> If so, Im forced to store my files in multiple Miners to decrease the chances of losing my backup.

yes you will have to consider that

>  But I will have to sleep thinking that all can be lost. To remove this fear I must use centralized Cloud Providers.

amazon can just delete your files as well. or google, ms, ...

> So, it is not worthy using FIlecoin at all. There has to be a way of making it profitable to Miners and at the same time give End Users relief that they can trust in this network and their files are safe, make it stupidly cheaper than AWS/Google Cloud/Azure/Alibaba Cloud, and finally as fast as those cloud provider are. Of course there will be trade-offs you will have to choose, or make it 100% visible so the End User can choose.

you have a wrong picture what filecoin is doing. filecoin itself does not store data - storage providers do at their own free will. the filecoin blockchain offers the fabric of payments, proof of storage +++ to both the data storage clients and the providers. Filecoin itself never sees the data - it only knows the data by hash. the storage operation itself is a 1:1 operation as you'd have it with centralized cloud providers.

**Reply by AllanOricil** [reply]
Centralized Cloud Providers can't say they won't accept my files If I pay them what they asked. Here Im seeing Providers not willing to store my files even when I did pay what they wanted ( I got what they wanted from this site https://plus.fil.org/miners). Here it is not a 1:1 as for my own experience.

Centralized Cloud Providers allow me to protect my files with replicas all around the world. I know i can do that with filecoin network by just asking multiple Providers to store my data. But I just don't want the Providers to not say No to me and having to spend lot of time knocking doors until one finally accepts my deal. As you pointed out, both centralized and decentralized services could lose all your data, but the risk in far higher in decentralized network as each provider would have its own way of protecting the data. So I would not say that this is a 1:1.

What Im trying to say is that you must make the Trade-offs clear and real to make it work.

**Reply by dd45e640b42e6da7da96faee3996ef7c** [reply]
i get your concerns and the problems with user experience and i agree - my point comes from the technical perspective and the tech does not support the expectations.

**Reply by AllanOricil** [reply]
This video https://filecoin.io/ shows that my expectations should be close (or almost equal) to what Centralized Cloud Providers do. If the Tech can't provide it to me, you can't say it can with this video. Im almost sure everybody will think the same as I did.

**Comment by jennijuju** [comment]
Hi @AllanOricil , thank for raising your questions and concerns. This specific tutorial you pointed out is for folks “try things  out and see the flow” using lotus and we should’ve added a callout to say , “for testing, use a testnet or devnet.

That being said, despite lotus implements a market today -  we don’t expect it to provide the best user experience to the end users. Lotus is more to expose the interface for application developers to build an intuitive and easy to use  UX for end users, which covers many pain points you mentioned: storage provider selection, data prep, replications and so on.

one of the best examples in the market right now would be http://estuary.tech/, check it out!

also - I think there are multiple better SP reputation and profile website that’s under development to make the SP selection experience better for clients. But for now, most direct way would be find a SP and work with them, for that you may post a request in this forum under “market” or joining #fil-deal-market in filecoin slack

**Reply by AllanOricil** [reply]
@jennijuju could your team give me access to this estuary.tech. I would like to start playing with it. I have already asked for access in the site form.

**Reply by jennijuju** [reply]
The estuary team (not lotus) is providing access to useful usecases - if you applied via the form, they will triage and get back to you.

**Comment by LexLuthr** [comment]
I am archiving this discussion as there has been no new updates from the author.
Please refer to https://github.com/filecoin-project/lotus/discussions/7228 for the discussion guidelines.
